client/thin-replica-client/src/trc_hash.cpp

```diff
@@ -68,40 +70,80 @@ static string hashUpdateFromEntryHashes(uint64_t block_id, const map<string, str
+    concatenated_entry_hashes.append((id_as_bytes + (sizeof(id) - i)), 1);
+  }
+#endif  // if BOOST_LITTLE_ENDIAN defined/else
+
+  for (const auto& hash : entry_hashes) {
+    concatenated_entry_hashes.append(hash);
   }
-  auto& legacy_event = std::get<Update>(update);
-  map<string, string> entry_hashes;
-  for (const auto& kvp : legacy_event.kv_pairs) {
-    string key_hash = ComputeSHA256Hash(kvp.first);
-    if (entry_hashes.count(key_hash) > 0) {
-      throw invalid_argument("hashUpdate called for an update that contains duplicate keys.");
+
+  return ComputeSHA256Hash(concatenated_entry_hashes);
+}
+
+string hashUpdate(const EventVariant& ev) {
+  if (std::holds_alternative<Update>(ev)) {
+    auto& legacy_event = std::get<Update>(ev);
+    map<string, string> entry_hashes;
+    for (const auto& kvp : legacy_event.kv_pairs) {
+      string key_hash = ComputeSHA256Hash(kvp.first);
+      if (entry_hashes.count(key_hash) > 0) {
+        throw invalid_argument("hashUpdate called for an update that contains duplicate keys.");
+      }
+      entry_hashes[key_hash] = ComputeSHA256Hash(kvp.second);
     }
-    entry_hashes[key_hash] = ComputeSHA256Hash(kvp.second);
+
+    return hashUpdateFromEntryHashes(legacy_event.block_id, entry_hashes);
   }
 
-  return hashUpdateFromEntryHashes(legacy_event.block_id, entry_hashes);
+  ConcordAssert(std::holds_alternative<EventGroup>(ev));
+  auto& event_group = std::get<EventGroup>(ev);
+  set<string> entry_hashes;
+  for (const auto& event : event_group.events) {
+    string event_hash = ComputeSHA256Hash(event);
+    ConcordAssert(entry_hashes.count(event_hash) < 1);
+    entry_hashes.emplace(event_hash);
+  }
+
+  return hashUpdateFromEntryHashes(event_group.id, entry_hashes);
 }
 
 string hashUpdate(const Data& update) {
-  if (update.has_event_group()) {
-    // TODO: Support Event Group hashes
-    return string();
-  }
-  ConcordAssert(update.has_events());
-  map<string, string> entry_hashes;
-  for (const auto& kvp : update.events().data()) {
-    string key_hash = ComputeSHA256Hash(kvp.key());
-    if (entry_hashes.count(key_hash) > 0) {
-      throw invalid_argument("hashUpdate called for an update that contains duplicate keys.");
+  if (update.has_events()) {
+    map<string, string> entry_hashes;
+    for (const auto& kvp : update.events().data()) {
+      string key_hash = ComputeSHA256Hash(kvp.key());
+      if (entry_hashes.count(key_hash) > 0) {
+        throw invalid_argument("hashUpdate called for an update that contains duplicate keys.");
+      }
+      entry_hashes[key_hash] = ComputeSHA256Hash(kvp.value());
     }
-    entry_hashes[key_hash] = ComputeSHA256Hash(kvp.value());
+
+    return hashUpdateFromEntryHashes(update.events().block_id(), entry_hashes);
+  }
+  ConcordAssert(update.has_event_group());
+  set<string> entry_hashes;
+  for (const auto& event : update.event_group().events()) {
+    string event_hash = ComputeSHA256Hash(event);
```

Comment:
Same comment as above, here.

Reply:
The event in this for-loop comes from EventGroup.events (thin_replica.proto) which is a repeated bytes field so essentially a `vector<string>` type. Tags shouldn't be included by the server.